downstairs/src/repair.rs

```diff
@@ -0,0 +1,386 @@
+}]
+async fn get_extent(
+    rqctx: Arc<RequestContext<FileServerContext>>,
+    path: Path<Eid>,
+) -> Result<Response<Body>, HttpError> {
+    let eid = path.into_inner().eid;
+    let extent_path = extent_path(rqctx.context().region_dir.clone(), eid);
+
+    get_file(extent_path).await
+}
+
+/**
+ * Stream the contents of the metadata .db file for the given extent ID
+ */
+#[endpoint {
+    method = GET,
+    path = "/extent/{eid}/db",
+    unpublished = false,
+}]
+async fn get_db(
+    rqctx: Arc<RequestContext<FileServerContext>>,
+    path: Path<Eid>,
+) -> Result<Response<Body>, HttpError> {
+    let eid = path.into_inner().eid;
+    let mut extent_path = extent_path(rqctx.context().region_dir.clone(), eid);
+    extent_path.set_extension("db");
+
+    get_file(extent_path).await
+}
+
+/**
+ * Stream the contents of the metadata .db-shm file for the given extent ID
+ */
+#[endpoint {
+    method = GET,
+    path = "/extent/{eid}/db-shm",
+    unpublished = false,
+}]
+async fn get_shm(
+    rqctx: Arc<RequestContext<FileServerContext>>,
+    path: Path<Eid>,
+) -> Result<Response<Body>, HttpError> {
+    let eid = path.into_inner().eid;
+    let mut extent_path = extent_path(rqctx.context().region_dir.clone(), eid);
+    extent_path.set_extension("db-shm");
+
+    get_file(extent_path).await
+}
+
+/**
+ * Stream the contents of the metadata .db-wal file for the given extent ID
+ */
+#[endpoint {
+    method = GET,
+    path = "/extent/{eid}/db-wal",
+    unpublished = false,
+}]
+async fn get_wal(
+    rqctx: Arc<RequestContext<FileServerContext>>,
+    path: Path<Eid>,
+) -> Result<Response<Body>, HttpError> {
+    let eid = path.into_inner().eid;
+    let mut extent_path = extent_path(rqctx.context().region_dir.clone(), eid);
+    extent_path.set_extension("db-wal");
+
+    get_file(extent_path).await
+}
+
+async fn get_file(path: PathBuf) -> Result<Response<Body>, HttpError> {
+    println!("Request for file {:?}", path);
+    /*
+     * Make sure our file is neither a link nor a directory.
+     */
+    let m = path
+        .symlink_metadata()
+        .map_err(|_| HttpError::for_bad_request(None, "ENOENT".to_string()))?;
+
+    if m.file_type().is_symlink() {
+        Err(HttpError::for_bad_request(None, "EMLINK".to_string()))
+    } else if path.is_dir() {
+        Err(HttpError::for_bad_request(None, "EBADF".to_string()))
+    } else {
+        let file = tokio::fs::File::open(&path).await.map_err(|_| {
+            HttpError::for_bad_request(None, "EBADF".to_string())
+        })?;
+
+        let file_stream = hyper_staticfile::FileBytesStream::new(file);
+        let content_type = "application/octet-stream".to_string();
+
+        Ok(Response::builder()
+            .status(StatusCode::OK)
+            .header(http::header::CONTENT_TYPE, content_type)
+            .body(file_stream.into_body())?)
+    }
+}
+
+#[derive(Deserialize, Serialize, JsonSchema)]
+struct ExtentFiles {
+    files: Vec<String>,
+}
```

Comment:
it's fine either way, but you could (I think??) just return `Vec<String>` directly rather than wrapping them in a struct